### PR TITLE
Updated model-archiver manifest

### LINKED ISSUE #325: Model archive manifest file version mismatch
The Model archive MANIFEST.json file should include correct version numbers for all components and have an additional timestamp field for the time when it was created.

Current manifest file for the vgg11.mar 

```
{
  "runtime": "python",
  "model": {
    "modelName": "vgg11",
    "serializedFile": "vgg11-bbd30ac9.pth",
    "handler": "image_classifier",
    "modelFile": "model.py",
    "modelVersion": "1.0"
  },
  "modelServerVersion": "1.0",
  "implementationVersion": "1.0",
  "specificationVersion": "1.0"
}
```

### PR DESCRIPTION
**Pull request by harshbafna**
## Description

 - added following new information in the manifest
   - createdOn  : date time for when the model archive was created
   - archiverVersion : torch-model-archiver version used for creating the model archive

 - removed following unused parameters from manifest
   - serverVersion
   - implementation_version
   - specification_version
   - engine
   - publisher

 - Cleaned up unused publisher and engine reference from frontend Java layer.
 - removed deprecation waring from backend python layer

Sample manifest file generated inside model archive : 
{
  "createdOn": "10/06/2020 12:30:56",
  "runtime": "python",
  "model": {
    "modelName": "densenet161",
    "serializedFile": "densenet161-8d451a50.pth",
    "handler": "image_classifier",
    "modelFile": "model.py",
    "modelVersion": "1.0"
  },
  "archiverVersion": "0.1.1"
}

Fixes #325

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Feature/Issue validation/testing

 - Create densenet161 mar using follwing doc : https://github.com/pytorch/serve#store-a-model
 - Unzip mar file
 - cat MAR-INF/MANIFEST.json
 - register and run inference on the newly created mar.

 - register any model from model_zoo and run inference on the same. [to ensure backward compatability]

[sanity script logs](https://github.com/pytorch/serve/files/4763011/sanity_script_logs.txt)
[regression_test_325.txt](https://github.com/pytorch/serve/files/4763199/regression_test_325.txt)

## Checklist:

- [x] Have you added tests that prove your fix is effective or that this feature works?
- [x] New and existing unit tests pass locally with these changes?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: frontend/modelarchive/src/main/java/org/pytorch/serve/archive/ModelArchive.java [hunk 1]
```diff
@@ -167,8 +167,14 @@ public void validate() throws InvalidModelException {
                 throw new InvalidModelException("Runtime is not defined or invalid.");
             }
 
-            if (manifest.getEngine() != null && manifest.getEngine().getEngineName() == null) {
-                throw new InvalidModelException("engineName is required in <engine>.");
+            if (manifest.getArchiverVersion() == null) {
```
Comment (chauhang) [inline frontend/modelarchive/src/main/java/org/pytorch/serve/archive/ModelArchive.java:170]: @harshbafna  Will people need to recreate their .mar files for existing models after this change? Want to understand the implications from migration perspective

File: model-archiver/model_archiver/tests/unit_tests/test_model_packaging_utils.py [hunk 2]
```diff
@@ -138,27 +136,17 @@ def __init__(self, **kwargs):
...
-        engine = EngineType.MXNET.value
         model_name = 'my-model'
         handler = 'a.py::my-awesome-func'
         serialized_file = 'model.pt'
         model_file = 'model.pt'
         version = "1.0"
         source_vocab = None
 
-        args = Namespace(author=author, email=email, engine=engine, model_name=model_name, handler=handler,
+        args = Namespace(author=author, email=email, model_name=model_name, handler=handler,
```
Comment (chauhang) [inline model-archiver/model_archiver/tests/unit_tests/test_model_packaging_utils.py:146]: @harshbafna How can user's specify the author/email? The CLI doesn't have any documented steps for this: https://github.com/pytorch/serve/blob/master/model-archiver/README.md#arguments

File: frontend/modelarchive/src/main/java/org/pytorch/serve/archive/Manifest.java [hunk 3]
```diff
@@ -1,44 +1,26 @@
...
 import com.google.gson.annotations.SerializedName;
-import java.util.LinkedHashMap;
-import java.util.Map;
 
 public class Manifest {
 
-    private String specificationVersion;
-    private String implementationVersion;
+    private String createdOn;
     private String description;
```
Comment (chauhang) [inline frontend/modelarchive/src/main/java/org/pytorch/serve/archive/Manifest.java:8]: @harshbafna How can user's specify description for the .mar files? At present there is no documented CLI option for this: https://github.com/pytorch/serve/blob/master/model-archiver/README.md#arguments

File: frontend/modelarchive/src/main/java/org/pytorch/serve/archive/ModelArchive.java (hunk 1 above)
Comment (harshbafna) [reply frontend/modelarchive/src/main/java/org/pytorch/serve/archive/ModelArchive.java:170]: @chauhang : Per current changes, users don't need to re-create their existing .mar files. However, added a warning to migrate to the latest version.

The sanity script and regression-suite execute successfully, which ensures that the old mar files from model-zoo work without any issues.

File: model-archiver/model_archiver/tests/unit_tests/test_model_packaging_utils.py (hunk 2 above)
Comment (harshbafna) [reply model-archiver/model_archiver/tests/unit_tests/test_model_packaging_utils.py]: Cleaned up unused params.

File: frontend/modelarchive/src/main/java/org/pytorch/serve/archive/Manifest.java (hunk 3 above)
Comment (harshbafna) [reply frontend/modelarchive/src/main/java/org/pytorch/serve/archive/Manifest.java:8]: Cleaned up unused params.

**Pull request merged**